src/server/routes/agree.ts

```diff
@@ -0,0 +1,99 @@
+import { Request } from 'express';
+import { rateLimitedTypedRouter as router } from '@/server/lib/typedRoutes';
+import { renderer } from '@/server/lib/renderer';
+import { ResponseWithRequestState } from '@/server/models/Express';
+import { read } from '../lib/idapi/user';
+import { logger } from '../lib/serverSideLogger';
+import { getConfiguration } from '../lib/getConfiguration';
+import { trackMetric } from '../lib/trackMetric';
+import deepmerge from 'deepmerge';
+import { addQueryParamsToUntypedPath } from '@/shared/lib/queryParams';
+import setupJobsUser from '../lib/jobs';
+
+const { defaultReturnUri } = getConfiguration();
+
+router.get(
+  '/agree/GRS',
+  async (req: Request, res: ResponseWithRequestState) => {
+    const SC_GU_U = req.cookies.SC_GU_U;
+    const state = res.locals;
+    const { returnUrl } = state.pageData;
```

Comment:
same as previous comment
```suggestion
    const { returnUrl } = state.queryParams;
```